File: src/main/java/com/jss/osiris/modules/osiris/beneficialOwner/service/BeneficialOwnerService.java
package com.jss.osiris.modules.osiris.beneficialOwner.service;

import java.util.List;

import com.jss.osiris.libs.exception.OsirisException;
import com.jss.osiris.modules.osiris.beneficialOwner.model.BeneficialOwner;
import com.jss.osiris.modules.osiris.quotation.model.Formalite;

public interface BeneficialOwnerService {

    public BeneficialOwner addOrUpdateBeneficialOwner(BeneficialOwner beneficialOwner);

    public BeneficialOwner addOrUpdateBeneficialOwnerToFormalite(BeneficialOwner beneficialOwner, Formalite formalite)
            throws OsirisException;

    public BeneficialOwner getBeneficialOwner(Integer id);

    public List<BeneficialOwner> getBeneficialOwnersByFormalite(Formalite formalite);

    public Boolean deleteBeneficialOwner(BeneficialOwner beneficialOwner);
}

File: client/src/app/modules/quotation/services/beneficial.owner.service.ts
import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { AppRestService } from "src/app/services/appRest.service";
import { BeneficialOwner } from '../model/beneficial-owner/BeneficialOwner';
import { Formalite } from "../model/Formalite";

@Injectable({
  providedIn: 'root'
})
export class BeneficialOwnerService extends AppRestService<BeneficialOwner> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  addOrUpdateBeneficialOwner(beneficialOwner: BeneficialOwner) {
    return this.postItem(new HttpParams(), "beneficial-owner", beneficialOwner);
  }
  getBeneficialOwners(formalite: Formalite) {
    return this.getList(new HttpParams().set("idFormalite", formalite.id), "beneficial-owners");
  }
  deleteBeneficialOwner(beneficialOwner: BeneficialOwner) {
    return this.get(new HttpParams().set("beneficialOwnerId", beneficialOwner.id), "beneficial-owner/delete");
  }
}